driver-core/src/test/unit/com/mongodb/internal/connection/InitialDnsSeedListDiscoveryProseTest.java

```diff
@@ -0,0 +1,147 @@
+/*
+ * Copyright 2008-present MongoDB, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.mongodb.internal.connection;
+
+import com.mongodb.MongoException;
+import com.mongodb.ServerAddress;
+import com.mongodb.connection.ClusterConnectionMode;
+import com.mongodb.connection.ClusterId;
+import com.mongodb.connection.ClusterSettings;
+import com.mongodb.connection.ClusterType;
+import com.mongodb.connection.ServerSettings;
+import com.mongodb.internal.dns.DefaultDnsResolver;
+import com.mongodb.internal.dns.DnsResolver;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.CsvSource;
+import org.mockito.Mock;
+
+import static java.util.Collections.singletonList;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.eq;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+import static org.mockito.MockitoAnnotations.openMocks;
+
+/**
+ * See https://github.com/mongodb/specifications/blob/master/source/initial-dns-seedlist-discovery/tests/README.md
+ */
+class InitialDnsSeedListDiscoveryProseTest {
+    private static final String SRV_SERVICE_NAME = "mongodb";
+
+    @Mock
+    private ClusterableServerFactory serverFactory;
+
+    @Mock
+    private DnsSrvRecordMonitorFactory dnsSrvRecordMonitorFactory;
+
+    private DnsMultiServerCluster cluster;
+
+    @BeforeEach
+    void setUp() throws Exception {
+        try (AutoCloseable ignored = openMocks(this)) {
+            when(serverFactory.getSettings()).thenReturn(ServerSettings.builder().build());
+            when(serverFactory.create(any(Cluster.class), any(ServerAddress.class))).thenReturn(mock(ClusterableServer.class));
```

Comment:
I have not seen `openMocks` before - is there a benefit to using this here, rather than using `Mockito.mock(` in `doTest`?

Reply:
It is the preferred counterpart (see https://javadoc.io/doc/org.mockito/mockito-core/3.4.0/org/mockito/MockitoAnnotations.html#openMocks-java.lang.Object-) of the previous common mockito usage as below:

```java
import org.mockito.MockitoAnnotations;

MockitoAnnotations.initMocks(this);
```

A better alternative is definitively the following test class level annotation:
```java

@ExtendWith(MockitoExtension.class)
```

but its usage currently requires our gradle dependency change.
How do you think which way to persue:
1) use `MockitoAnnotations.initMocks(this)`
2) add MockitoExtension related testing scope dependency and use `@ExtendWith(MockitoExtension.class)`
3) current status quo, which avoids deprecation warning but ends up with lots of confusion

Reply:
By "current status quo" in point 3, are you referring to `Mockito.mock(`? Could you elaborate on what the confusion is?

Reply:
It means the current approach as below:
```
try (AutoCloseable ignored = openMocks(this)) {
    // mock behaviour
}
```

It is a programmatic way to ensure each field annotation with `@Mock` has been gone through `Mockito.mock()` so we could focus on mocking their behavour in the try block.

All the three approaches are for the same purposes so we don't need to invoke `Mockito.mock()` manually one by one (which is tedious, less readable and error-prone).

Reply:
As we discussed, I reverted back Mock annotation usage to manually mock inside method.